### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by JantuDeb**
# Authentication
- Signup
- Login 
- Logout
### Refractored Category component
- Added context
image
image
image

### CODE REVIEW

File: src/components/navbar/Navbar.jsx [hunk 1]
```diff
@@ -22,6 +25,11 @@ const Navbar = () => {
           />
         </IconWrapper>
         <div className="flex">
+          <div className="profile">
+            {
+              authState.isLogedIn ? <button className="btn-grad-red radius-md py-1 px-2" onClick={()=>logOut()}>Logout</button> : <Link to="/login" className="btn-grad-red radius-md py-1 px-2 flex center">Login</Link>
```
Comment (Mohitkumar33) [inline src/components/navbar/Navbar.jsx]: an additional thing you can do is you can get firstName from the user which we get when we get the token and you can display "Hello , name of the user" on right of the logout button

**Review by Mohitkumar33 — APPROVED** [review]
Clean code

File: src/components/navbar/Navbar.jsx (hunk 1 above)
Comment (JantuDeb) [reply src/components/navbar/Navbar.jsx]: I will  do it, for now I'm testing logout functionality

**Review by Bhakti199 — APPROVED** [review]
This code looks fine to me.
Proper hierarchy is maintained.
This PR is good to merge.

**Pull request merged**